src-electron/db/query-impexp.js

```diff
@@ -674,7 +678,18 @@ WHERE
   } else if (reportingPolicy == dbEnums.reportingPolicy.prohibited) {
     attribute.reportable = false
   }
-
+  if (attributeId) {
+    forcedExternal = await queryUpgrade.getForcedExternalStorage(
+      db,
+      attributeId
+    )
+    storagePolicy = await queryUpgrade.computeStorageImport(
```

Comment:
In the function definition of computeStorageImport you retrieve the attribute name from attribute Id. Instead of doing that update the above query `SELECT
  A.ATTRIBUTE_ID,
  A.REPORTING_POLICY,
  A.STORAGE_POLICY
FROM
  ATTRIBUTE AS A
INNER JOIN
  ENDPOINT_TYPE_CLUSTER AS ETC
ON
  ETC.CLUSTER_REF = A.CLUSTER_REF OR A.CLUSTER_REF IS NULL
WHERE
  A.CODE = ?
  AND A.PACKAGE_REF IN (${dbApi.toInClause(packageIds)})
  AND A.SIDE = ETC.SIDE
  AND ETC.ENDPOINT_TYPE_CLUSTER_ID = ?
  AND ${
    attribute.mfgCode == null
      ? 'A.MANUFACTURER_CODE IS NULL'
      : 'A.MANUFACTURER_CODE = ?'
  }`` with A.NAME and do attributeName = atRow[0].NAME

Reply:
right but now we cannot reuse "computeStorageImport " because it will assume you already found all of that data. I think that it should only be used once though so it makes sense.

Reply:
done